src/Raven.Server/Web/System/TcpConnectionInfoHandler.cs

```diff
@@ -19,5 +30,139 @@ public Task Get()
 
             return Task.CompletedTask;
         }
+
+        [RavenAction("/info/remote-task/topology", "GET", AuthorizationStatus.RestrictedAccess)]
+        public Task GetRemoteTaskTopology()
+        {
+            var database = GetStringQueryString("database");
+            var databaseGroupId = GetStringQueryString("groupId");
+            var remoteTask = GetStringQueryString("remote-task");
+
+            Authenticate(HttpContext, ServerStore, database, remoteTask);
+
+            List<string> nodes;
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var pullReplication = ServerStore.Cluster.ReadPullReplicationDefinition(database, remoteTask, context);
+                var topology = ServerStore.Cluster.ReadDatabaseTopology(context, database);
+                nodes = GetResponsibleNodes(topology, databaseGroupId, pullReplication);
+            }
+
+            using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                var output = new DynamicJsonArray();
+                var clusterTopology = ServerStore.GetClusterTopology();
+                foreach (var node in nodes)
+                {
+                    output.Add(clusterTopology.GetUrlFromTag(node));
+                }
+                context.Write(writer, new DynamicJsonValue
+                {
+                    ["Results"] = output
+                });
+            }
+            return Task.CompletedTask;
+        }
+
+        [RavenAction("/info/remote-task/tcp", "GET", AuthorizationStatus.RestrictedAccess)]
+        public Task GetRemoteTaskTcp()
+        {
+            var remoteTask = GetStringQueryString("remote-task");
+            var database = GetStringQueryString("database");
+
+            Authenticate(HttpContext, ServerStore, database, remoteTask);
+
+            using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                var output = Server.ServerStore.GetTcpInfoAndCertificates(HttpContext.Request.GetClientRequestedNodeUrl());
+                context.Write(writer, output);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        public static void Authenticate(HttpContext httpContext, ServerStore serverStore, string database, string remoteTask)
+        {
+            var feature = httpContext.Features.Get<IHttpAuthenticationFeature>() as RavenServer.AuthenticateConnection;
+
+            if (feature == null) // we are not using HTTPS 
+                return;
+
+            switch (feature.Status)
+            {
+                case RavenServer.AuthenticationStatus.Allowed:
```

Comment:
This isn't true. The certificate is allowed to _some_ database, not necessarily this one.